libs/ledger/src/chain/main_chain.cpp

```diff
@@ -621,6 +630,24 @@ bool MainChain::GetPathToCommonAncestor(Blocks &blocks, BlockHash tip, BlockHash
   return success;
 }
 
+/**
+ * Return whether, for a given block number and weight, whether different blocks with the same
+ * weight have been seen
+ *
+ * @param block_number Block number being queried
+ * @param block_weight Weight being queried
+ * @return Whether pair (block_number, block_weight) pair correspond to stutter blocks
+ */
+bool MainChain::IsStutterBlock(BlockNumber block_number, BlockWeight block_weight) const
+{
+  if (stutter_blocks_.find(block_number) == stutter_blocks_.end() ||
```

Comment:
I don't tend to like creating empty entries like this in the maps, but can change it.

Reply:
Actually, changing this to use [] instead of find breaks the fact that the function does not alter the main chain. Since this function is used to query the main chain from consensus I am leaving it as it is.